pages/index.js

```diff
@@ -1,59 +1,87 @@
+//import { Router } from "next/router";
 
 export default function HomePage() {
-  const [selectedLanguage, setSelectedLanguage] = useState("English");
-  const [selectedOptions, setSelectedOptions] = useState(
-    "Simple Sentences & For Kids"
-  );
-  const [selectedLevels, setSelectedLevels] = useState("Basic");
-  //const [frenchText, setFrenchText] = useState("");
+  const [config, setConfig] = useState({
+    language: "",
+    theme: "",
+    difficulty: "",
+  });
 
-  function handleNext() {
-    /*  fetch("french.json")
-      .then((res) => res.json())
-      .then((data) => console.log(data));*/
-  }
+  /* // call from json file
+  const selectedTwister = twisters.find((twister) => {
+    return (
+      twister.language === config.language &&
+      twister.theme === config.theme &&
+      twister.difficulty === config.difficulty
+    );
+  });*/
+  // call from openai
+  const [answer, setAnswer] = useState(null);
+  const [loading, setLoading] = useState(false);
 
-  function handleOption(option) {
-    setSelectedOptions(option);
-    console.log("option est " + option);
-  }
-  function handLeLevel(level) {
-    setSelectedLevels(level);
-    console.log("  Level est" + level);
-  }
+  async function fetcher(data) {
+    setLoading(true);
 
-  function handleLanguage(language) {
-    setSelectedLanguage(language);
-    console.log(" language est " + language);
+    try {
+      const response = await fetch("/api/gpt", {
+        method: "POST",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify(data),
+      });
+      if (response.ok) {
+        const result = await response.json();
+        setAnswer(result);
+        console.log(result);
+        /*   Router.push({
+          pathname: "/twisterPage",
+          query: {
+            answer: result.answer.content 
+          },
+        });*/
+      } else {
+        console.error("Bad Response");
+      }
+    } catch (error) {
+      console.error(error);
+    } finally {
+      setLoading(false);
+    }
+  }
+  function handleSubmit(event) {
+    event.preventDefault();
+    fetcher(config);
   }
 
   return (
     <>
       <ContainerMain>
-        <h2> Homepage</h2>
-        <h2> Text Language </h2>
-        <PageLanguage handleLanguage={handleLanguage} />
-        <h2> Text Option</h2>
-        <PageOptions handleOption={handleOption} />
-        <h2> Text Level </h2>
-        <PageLevel handLeLevel={handLeLevel} />
-        <StyledLink href={"/folgePage"} onClick={handleNext}>
-          {" "}
-          Next
-        </StyledLink>
+        <h2> tongue twister</h2>
+        <label htmlFor="lang-select">Language:</label>
+        <Languages config={config} setConfig={setConfig} />
+        <label htmlFor="lang-select">Theme:</label>
+        <Themes config={config} setConfig={setConfig} />
+        <label htmlFor="lang-select">Difficulty:</label>
+        <Difficulty config={config} setConfig={setConfig} />
```

Comment:
Do not pass set functions directly as props into components

wrap you set functions in handler functions

```javascript
function handleSomething(){
setSomething(...)
}
``` 

Call your props `onSomething`

```javascript
<Component onSomething={handleSomething}/>
```

This is called Lifting up State and you find information on this topic within our React State Handouts